docs/catalog.py

```diff
@@ -22,6 +22,31 @@ def custom_walk(top):
             yield entry
 
 
+def all_subtypes_of_artifact(artifact):
+    if (
+        artifact is None
+        or isinstance(artifact, str)
+        or isinstance(artifact, bool)
+        or isinstance(artifact, int)
+        or isinstance(artifact, float)
+    ):
+        return []
+    if isinstance(artifact, list):
+        to_return = []
+        for art in artifact:
+            to_return.extend(all_subtypes_of_artifact(art))
+        return to_return
+    # artifact is a dict
+    to_return = []
+    for key, value in artifact.items():
+        if isinstance(value, str):
+            if key == "type":
```

Comment:
just from reading the code i am not sure what is going on .. how can `artifact` become a dict? if its passed a dict in the first call? what does `all_subtypes_of_artifact` mean? maybe a docstring may help ..